pdbApp/pdbgroup.cpp

```diff
@@ -349,12 +350,24 @@ void PDBGroupPut::put(pvd::PVStructure::shared_pointer const & value,
     // assume value may be a different struct each time... lot of wasted prep work
     const size_t npvs = channel->pv->members.size();
     std::vector<std::tr1::shared_ptr<PVIF> > putpvif(npvs);
+    std::vector<void *> asWritePvt;
```

Comment:
Where possible, I try to write exception safe code.  This is relevant here as calls like `ret |= putpvif[i]->get(*changed, doProc);` below can and do throw exceptions.

One solution to avoid this would be to define a new type to own asWrite pointers.  eg.

```c++
struct AsWritePvt {
    void *pvt;
    explicit AsWritePvt(void* pvt): pvt(pvt) {}
    ~AsWritePvt() {
        asTrapWriteAfterWrite(pvt);
    }
};
```

And then define `std::vector<AsWritePvt>`.

Reply:
Good point,

Where do you prefer this new structure to be defined? It should be in one of the headers that's included in both `pdbsignle.cpp` and `pdbgroup.cpp` as I would use the same approach on both places. Maybe `pdb.h`?